planning/route_handler/src/route_handler.cpp

```diff
@@ -880,8 +893,38 @@ bool RouteHandler::getClosestLaneletWithConstrainsWithinRoute(
     route_lanelets_, search_pose, closest_lanelet, dist_threshold, yaw_threshold);
 }
 
-bool RouteHandler::getNextLaneletWithinRoute(
-  const lanelet::ConstLanelet & lanelet, lanelet::ConstLanelet * next_lanelet) const
+boost::optional<lanelet::ConstLanelet> RouteHandler::getClosestRouteLaneletFromCurrent(
+  const Pose & search_pose, const lanelet::ConstLanelet & current_lanelet,
+  const double dist_threshold, const double yaw_threshold) const
+{
+  boost::optional<lanelet::ConstLanelet> closest_lanelet = {};
+
+  lanelet::ConstLanelets previous_lanelets, next_lanelets, lanelet_sequence;
+  if (getPreviousLaneletsWithinRoute(current_lanelet, &previous_lanelets)) {
+    lanelet_sequence = previous_lanelets;
+  }
+
+  lanelet_sequence.push_back(current_lanelet);
+  const auto & current_neighbors =
+    lanelet::utils::query::getAllNeighbors(routing_graph_ptr_, current_lanelet);
```

Comment:
Neighbor lanelets may not be in the route. Is this okay ?

Reply:
you are right, will fix